torch/distributed/checkpoint/_fsspec_filesystem.py

```diff
@@ -294,7 +297,7 @@ def _write_files_from_queue(
                     )
 
                 for tensor, write_item in loader.values():
-                    assert not tensor.is_cuda
+                    assert tensor.device.type not in ["cuda", torch._C._get_privateuse1_backend_name()]
```

Comment:
nit: maybe change it to `assert tensor.is_cpu` as that's what we need to ensure.